src/features/marketplace/store/marketplace.slice.ts

```diff
@@ -1,25 +1,48 @@
 import { createSlice } from '@reduxjs/toolkit';
 import type { RootState } from '../../../store/types';
-import { getOwner } from './marketplace.actions';
+import { Item } from '../../../types/item';
+import ItemStatus from '../process/enums/item-status.enum';
+import { getOwner, listItem } from './marketplace.actions';
 
 export type MarketplaceState = {
+  isLoading: boolean;
   owner: string | undefined;
 };
 
 const initialState: MarketplaceState = {
   owner: '',
+  isLoading: false,
 };
 
 const marketplaceSlice = createSlice({
   name: 'marketplace',
   initialState,
   reducers: {},
   extraReducers: (builder) => {
+    builder.addCase(listItem.pending, (state) => {
+      state.isLoading = true;
+    });
+    builder.addCase(listItem.fulfilled, (state) => {
+      state.isLoading = false;
+    });
+    builder.addCase(listItem.rejected, (state) => {
+      state.isLoading = false;
+    });
     builder.addCase(getOwner.fulfilled, (state, { payload }) => {
       state.owner = payload;
     });
   },
 });
 
 export const selectOwner = (state: RootState) => state.marketplace.owner;
+
+export const selectCanIList = (state: RootState, item: Item) => {
+  const { address } = state.auth;
+
+  if (item.owner.address === address) return true;
+  if (item.status === ItemStatus.NotListed) return true;
```

Comment:
This is not correct. What you are saying is 'if the user is the owner, he can list. If he is not the owner, but the item is not listed, the user will be able to listed anyways'. The correct way would be `return item.owner.address === address && item.status === ItemStatus.NotListed`